packages/graphback-core/src/runtime/GraphbackCRUDService.ts

```diff
@@ -23,44 +23,48 @@ export interface GraphbackCRUDService<Type = any, GraphbackContext = any> {
    * Implementation for object creation
    *
    * @param data input data
+   * @param fields fields to select from datasource after creation
    * @param context context object passed from graphql or rest layer
    */
-  create(data: Type, context?: GraphbackContext): Promise<Type>;
+  create(data: Type, fields: string[], ccontext?: GraphbackContext): Promise<Type>;
 
   /**
    * Implementation for object updates
    *
    * @param data input data including id
+   * @param fields fields to select from datasource after update
    * @param context context object passed from graphql or rest layer
    */
-  update(data: Type, context?: GraphbackContext): Promise<Type>;
+  update(data: Type,fields: string[], context?: GraphbackContext): Promise<Type>;
 
   /**
    * Implementation for object deletes
    *
-   * @param id of the object to delete
    * @param data data used for consistency reasons
+   * @param fields fields to select from datasource after deletion
    * @param context context object passed from graphql or rest layer
    */
-  delete(data: Type, context?: GraphbackContext): Promise<Type>;
+  delete(data: Type, fields: string[], ccontext?: GraphbackContext): Promise<Type>;
 
   /**
    * Fetch a single record by its unique attribute(s)
    *
    * @param filter - the unique attributes to fetch the record with
+   * @param fields fields to select from datasource
    * @param context context object from GraphQL/REST layer
    */
-  findOne(filter: Type, context?: GraphbackContext): Promise<Type>;
+  findOne(filter: Type, fields: string[], ccontext?: GraphbackContext): Promise<Type>;
```

Comment:
`ccontext` :)


Also -1 for adding ` fields: string[]` it brings negative pattern to break API and increases number of arguments.

There are two options here:

1) Introduce options object (we had that before for subscriptions flags but since then we use context instead
2) use `context.graphbackOptions` or something similar and make context non optional

No strong opinion but utilizing context will setup good pattern for possible extensions/requirements in future. Current implementation will encourage constant breaking when we need yet another thing to add

Reply:
Also my plan to to get early feedback on the api design. This is really useful feedback that I'll look to bring on later and mark the PR ready for review.